frontends/main/src/app-pages/DashboardPage/CoursewareDisplay/DashboardCard.test.tsx

```diff
@@ -6,67 +6,148 @@ import { faker } from "@faker-js/faker/locale/en"
 import moment from "moment"
 import { EnrollmentMode, EnrollmentStatus } from "./types"
 
-describe("EnrollmentCard", () => {
-  test("It shows course title with link to marketing url", () => {
-    const course = dashboardCourse()
-    renderWithProviders(<DashboardCard dashboardResource={course} />)
-
-    const courseLinks = screen.getAllByRole("link", {
-      name: course.title,
-    })
-    for (const courseLink of courseLinks) {
-      expect(courseLink).toHaveAttribute("href", course.marketingUrl)
-    }
-  })
-
-  test("Courseware button is disabled if course has not started", () => {
-    const course = dashboardCourse({
+const pastDashboardCourse: typeof dashboardCourse = (...overrides) => {
+  return dashboardCourse(
+    {
       run: {
-        startDate: faker.date.future().toISOString(),
+        startDate: faker.date.past().toISOString(),
+        endDate: faker.date.past().toISOString(),
       },
-    })
-    renderWithProviders(<DashboardCard dashboardResource={course} />)
-    const coursewareButtons = screen.getAllByRole("button", {
-      name: "Continue Course",
-      hidden: true,
-    })
-    for (const coursewareButton of coursewareButtons) {
-      expect(coursewareButton).toBeDisabled()
-    }
-  })
-
-  test("Courseware button is enabled if course has started AND NOT ended", () => {
-    const course = dashboardCourse({
+    },
+    ...overrides,
+  )
+}
+const currentDashboardCourse: typeof dashboardCourse = (...overrides) => {
+  return dashboardCourse(
+    {
       run: {
         startDate: faker.date.past().toISOString(),
         endDate: faker.date.future().toISOString(),
       },
-    })
-    renderWithProviders(<DashboardCard dashboardResource={course} />)
-    const coursewareLinks = screen.getAllByRole("link", {
-      name: "Continue Course",
-    })
-    for (const coursewareLink of coursewareLinks) {
-      expect(coursewareLink).toHaveAttribute("href", course.run.coursewareUrl)
-    }
-  })
-
-  test("Courseware button says 'View Course' if course has ended", () => {
-    const course = dashboardCourse({
+    },
+    ...overrides,
+  )
+}
+const futureDashboardCourse: typeof dashboardCourse = (...overrides) => {
+  return dashboardCourse(
+    {
       run: {
-        startDate: faker.date.past().toISOString(),
-        endDate: faker.date.past().toISOString(),
+        startDate: faker.date.future().toISOString(),
+        endDate: faker.date.future().toISOString(),
       },
-    })
+    },
+    ...overrides,
+  )
+}
+
+describe.each([
+  { display: "desktop", testId: "enrollment-card-desktop" },
+  { display: "mobile", testId: "enrollment-card-mobile" },
+])("EnrollmentCard $display", ({ testId }) => {
```

Comment:
@gumaerc I did decide to parameterize the desktop vs mobile testing. I'm worried without this we'll forget to test both cases in the future.

(Not that the possibility of forgetting is gone now. You could still ignore the parameterized testId.)